apps/examples/wifi_manager_sample/wm_test_on_off.c

```diff
@@ -41,26 +41,23 @@
 #define WIFIMGR_AUTH WIFI_MANAGER_AUTH_WPA2_PSK
 #define WIFIMGR_CRYPTO WIFI_MANAGER_CRYPTO_AES
 
-
-#define WM_TEST_SIGNAL										\
-	do {													\
-		pthread_mutex_lock(&g_wm_mutex);					\
-		printf("T%d send signal\n", getpid());	\
-		pthread_cond_signal(&g_wm_cond);					\
-		pthread_mutex_unlock(&g_wm_mutex);					\
+static sem_t g_wm_sem = SEM_INITIALIZER(0);
+#define WM_TEST_SIGNAL								\
+	do {											\
+		sem_post(&g_wm_sem);						\
+		printf("[WO] T%d send signal\t %s:%d\n", getpid(), __FUNCTION__, __LINE__); \
```

Comment:
We'd better print the log before calling `sem_post()`, because as soon as called `sem_post()`, context can be changed to waited task/pthread, so log can be mixed.